source/agora/utils/Test.d

```diff
@@ -627,106 +620,413 @@ unittest
 
 /*******************************************************************************
 
-    Generate a new transaction that evenly splits the input accross parties
+    A reference to an `Output` within a `Transaction`
 
-    The `input` transaction will be split evenly in `toward.length` outputs,
-    each of which will be controlled by a key in `toward`.
-    If the sum of outputs in the transaction is not a multiple of
-    `toward.length`, the leftover `Amount` will be added to
-    the output for `toward[0]`.
-    Like other testing utilities, if an error happens (e.g. `input` is invalid,
-    or `from` is missing some keys), an `assert` will be triggered.
+    In order to reference an `Output` in an UTXO, one need to know the hash
+    of the transaction, as well as the index. This structure holds both index
+    and `Transaction` for convenience.
 
-    Params:
-        type = Type of transaction to generate (frozen or simple payment)
-        input = Transaction to spend completely
-        from = Array of keys controlling all the outputs in `input`
-        toward = Receivers for the newly-created output
+*******************************************************************************/
 
-    Returns:
-        A newly created, valid, and signed `Transaction` spending `input`.
+public struct OutputRef
+{
+    ///
+    public Hash hash () const nothrow @safe
+    {
+        return hashFull(this.tx);
+    }
+
+    ///
+    public const(Output) output () const @safe pure nothrow @nogc
+    {
+        return this.tx.outputs[index];
+    }
+
+    /// The `Transaction` with the `Output` being referenced
+    public const(Transaction) tx;
+
+    /// The index of the `Output` being referenced
+    public uint index;
+}
+
+/*******************************************************************************
+
+    An helper utility to build transactions
+
+    This utility exposes a mutable transaction builder, used to simplify
+    generating complex `Transaction`s, as well as ensuring that generated
+    transactions are valid. Generating invalid `Transaction`s is not supported,
+    however one can generate a valid `Transaction` and mutate it afterwards.
+
+    Basics:
+    When building a transaction, one must first attach an `Output`,
```

Comment:
I don't see `Output` being supported in the constructors. And also the ctor taking in a public key is not mentioned. Or is that what you meant by Output? Hmm

Reply:
"Attaching an Output" doesn't mean that it is done via ctor. You could do `TxBuilder(key).attach(tx)`.
Also `Output` is not directly supported because we need a reference to the transaction, hence to give an `Output` you actually give a `Transaction` + index.